### ISSUE TITLE
Question about aggregating labels in your paper

### ISSUE THREAD

**Issue by HowieMa** [zlai0/MAST#2]
Hi, thank you for sharing such an awesome project. I have some questions about the details for your paper.

In the step 4 of Algorithm 1 (Section 3.3.1), you said "the output pixel's labels is determined by aggregating the labels of the ROI pixels". My question is how you treat several frames' ROI. 

Do you use the algorithm similar as [STM](http://openaccess.thecvf.com/content_ICCV_2019/papers/Oh_Video_Object_Segmentation_Using_Space-Time_Memory_Networks_ICCV_2019_paper.pdf). For example, if the Query's size is H * W * C (C is the number of channels), and the size of your restricted attention in all previous T Keys is P. Do you calculate an affinity matrix with size (HW * TPP)?
Or you just do the same thing as [Cycle-consistency](https://arxiv.org/abs/1903.07593). That is, a K-NN strategy and it average all previous predictions. 

I really appreciate it if you could solve my concerns. Looking forward to your official code. Thank you!

**Comment by zlai0** [comment]
Hi, sorry for the late reply. That's exactly right. We use the algorithm similar to STM. All attention areas are treated as one single large area (instead of T areas) and they should add up to 1 after softmax).

**Comment by HowieMa** [comment]
Thank you for the detailed answer, your answer and code inspire me a lot! 
May I ask another question? 
It seems that your [training code](https://github.com/zlai0/MAST/blob/master/main.py#L184) only has the first training step, as in your paper "_we first pretrain the network with a pair of input frames_". Thus could you please share the code that for _finetuning the model with multiple reference frames_? 
Look forward to your reply, thanks!

**Comment by zlai0** [comment]
Thanks for the interest. That's right - I did not add the code because we recently find that the finetuning step is not necessarily needed and the performance increase from it seems just marginal. I was not sure if I should over complicate the code. But yes, I could share the fine-tuning code. It should be pretty simple to implement as well.

**Comment by houhouhouhou11** [comment]
@zlai0 Could you share the  code that for "finetuning the model with multiple reference frames"? When I train the multiple reference frames, the performance is similar to the single frame. Thanks
